### PR TITLE
⬆️ Update TypeScript

### PR DESCRIPTION
**Pull request by KATT**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,12 +64,15 @@
...
+    "tsdx": "^0.14.1",
+    "typescript": "^4.1.3"
   },
   "dependencies": {
     "lodash": "^4.17.20",
     "lodash-es": "^4.17.15"
+  },
+  "resolutions": {
+    "**/@typescript-eslint/eslint-plugin": "^4.1.1",
+    "**/@typescript-eslint/parser": "^4.1.1"
```
Comment (KATT) [inline package.json:76]: https://stackoverflow.com/a/63919395/590396

File: src/index.test.ts [hunk 2]
```diff
@@ -689,7 +689,7 @@ describe('stringify & parse', () => {
 
   test('regression #65: BigInt on Safari v13', () => {
     const oldBigInt = global.BigInt;
-    delete global.BigInt;
+    delete (global as any).BigInt;
```
Comment (KATT) [inline src/index.test.ts:692]: new in ts > 4
> The operand of a 'delete' operator must be optional

File: package.json (hunk 1 above)
Comment (Skn0tt) [reply package.json:76]: I'd like to stay on the old version of `tsdx` until this workaround isn't required any longer.

Or are there any new features in TSC 4.1 that SuperJSON would profit from?

File: src/index.test.ts (hunk 2 above)
Comment (Skn0tt) [reply src/index.test.ts:692]: Nice catch!

File: package.json (hunk 1 above)
Comment (KATT) [reply package.json:76]: Fair enough

**Pull request closed without merging**